### PR TITLE
Matrix multiplication and transpose optimize 

### PR DESCRIPTION
**Pull request by toor1245**

### CODE REVIEW

File: src/MatrixDotNet/NetCore/MatrixAsFixedBuffer.cs [hunk 1]
```diff
@@ -350,7 +363,10 @@ public override string ToString()
                 for (var i = 0; i < Rows; i++)
                 {
                     var span = span1.Slice(i * Columns, Columns);
-                    foreach (var t in span) builder.Append(t + " ");
+                    foreach (var t in span)
+                    {
+                        builder.Append(t + " ");
```
Comment (Dilorfin) [inline src/MatrixDotNet/NetCore/MatrixAsFixedBuffer.cs:368]: Are you joking?
Replace with two `builder.Append` calls....

File: src/MatrixDotNet/Vectorization/VectorBasic.cs [hunk 2]
```diff
@@ -54,5 +55,139 @@ public static unsafe Matrix<T> TensorProduct<T>(Vector<T> va, Vector<T> vb)
...
+        {
+            int len = va.Length;
+
+            if (len != vb.Length)
+            {
+                throw new MatrixDotNetException("vectors are not equal");
+            }
+
+            Vector<T> vc = new Vector<T>(len);
+            int i = 0;
```
Comment (Dilorfin) [inline src/MatrixDotNet/Vectorization/VectorBasic.cs:78]: Why this variable cannot be initialized just for `for` cycles?

File: src/MatrixDotNet/NetCore/MatrixAsFixedBuffer.cs [hunk 3]
```diff
@@ -269,7 +273,10 @@ public static MatrixAsFixedBuffer SubByRef(ref MatrixAsFixedBuffer left, ref Mat
         /// </exception>
         public static MatrixAsFixedBuffer MulByRef(ref MatrixAsFixedBuffer left, ref MatrixAsFixedBuffer right)
         {
-            if (left.Columns != right.Rows) throw new MatrixDotNetException("");
+            if (left.Columns != right.Rows)
+            {
+                throw new MatrixDotNetException("");
```
Comment (Dilorfin) [inline src/MatrixDotNet/NetCore/MatrixAsFixedBuffer.cs:278]: May be exception message should be more informative?)

Comment (toor1245) [reply src/MatrixDotNet/NetCore/MatrixAsFixedBuffer.cs:278]: ooooooooops)

**Review by Dilorfin — APPROVED** [review]

**Pull request merged**